sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DDSketchAggregator.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.aggregator;
+
+import com.datadoghq.sketch.ddsketch.SignedDDSketch;
+import com.datadoghq.sketch.ddsketch.mapping.CubicallyInterpolatedMapping;
+import com.datadoghq.sketch.ddsketch.store.UnboundedSizeDenseStore;
+import io.opentelemetry.api.common.Labels;
+import io.opentelemetry.sdk.metrics.data.MetricData.SummaryPoint;
+import io.opentelemetry.sdk.metrics.data.MetricData.ValueAtPercentile;
+import java.util.Arrays;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.function.Supplier;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public final class DDSketchAggregator extends AbstractAggregator {
+  static final double PRECISION = 0.001;
+
+  private static final AggregatorFactory AGGREGATOR_FACTORY =
+      () ->
+          new DDSketchAggregator(
+              () ->
+                  new SignedDDSketch(
+                      new CubicallyInterpolatedMapping(PRECISION), UnboundedSizeDenseStore::new));
+
+  public static AggregatorFactory getBalancedFactory() {
+    return AGGREGATOR_FACTORY;
+  }
+
+  // Since DDSketch is not thread safe, this queue is used to buffer calls to record, reducing
+  // the need to take a lock for every call. (With the downside of having to box the double.)
+  private final ArrayBlockingQueue<Double> pendingValues = new ArrayBlockingQueue<>(64);
```

Comment:
Doesn't a blocking queue also imply a taking a lock on every call? (it's a blocking queue :) ). 

https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/concurrent/ArrayBlockingQueue.java#L344

Is the idea that it would be less time under lock than if always calling `current.accept`?